lib/stateful_enum/active_record_extension.rb

```diff
@@ -16,7 +16,9 @@ def enum(definitions, &block)
       if block
         definitions.each_key do |column|
           states = enum[column]
-          (@_defined_stateful_enums ||= []) << StatefulEnum::Machine.new(self, column, (states.is_a?(Hash) ? states.keys : states), prefix, suffix, &block)
+          (@_defined_stateful_enums ||= {}).tap do |hash|
```

Comment:
I changed this to a `Hash` so it's possible to do this later to get `StatefulEnum::Machine` instances

```ruby
StateInspector.new(self.class.stateful_enum[column.to_sym], self)
```